Test/TestCase/SearchManagerTest.php

```diff
@@ -158,12 +159,49 @@ public function testRemove()
         $this->assertNull($view);
     }
 
+    public function testRemoveAll()
+    {
+        $this->createData();
+        $searchManager = $this->getSearchManager();
+        $viewRepo = $searchManager->getRepository(View::class);
+
+        /** @var View[] $views */
+        $views = array();
+        $count = 0;
+        while($count < 5) {
+            $view = new View();
+            $view->setId(uniqid());
+            $view->setBrowser('safari');
+            $view->setDevice('ios');
+            $view->setViews(10);
+            $view->setTags(array(new Tag('pro', 10.0)));
+            $view->setDate(new \DateTime('c'));
+            $viewRepo->save($view, true);
+            $views[] = $view;
+            $count++;
+        }
+
+        foreach($views as $originalView) {
+            $view = $viewRepo->findOneBy(array('id' => $originalView->getId()));
+            $this->assertNotNull($view);
+        }
+
+        $searchManager->getClient()->removeAll($searchManager->getClassMetadata(View::class));
```

Comment:
Expose the `removeAll()` in the SearchManager itself, and it should just take and index name and optionally type.